petpal/shelters/views.py

```diff
@@ -24,13 +69,42 @@ def shelters_list_view(request):
 PERMISSION:
 SUCCESS:
 '''
-@api_view(['GET', 'PUT', 'DELETE'])
+
+
+@api_view(['GET', 'PUT'])
+@permission_classes([IsAuthenticated])
 def shelter_detail_view(request, account_id):
+    # Check that the shelter exists
+    try:
+        shelter = Shelter.objects.get(id=account_id)
+    except Shelter.DoesNotExist:
+        return Response({'error': 'Shelter does not exist'}, status=status.HTTP_404_NOT_FOUND)
+
+    if shelter.role != PetPalUser.Role.SHELTER:
+        return Response({'error': 'Shelter does not exist'}, status=status.HTTP_404_NOT_FOUND)
+
+    # METHODS
     if request.method == 'GET':
-        return Response({'msg': 'shelter detail'}, status=status.HTTP_200_OK)
+        # Give details of the shelter
+        serializer = ShelterSerializer(shelter, many=False)
+        return Response({'msg': 'Shelter Detail', 'data': serializer.data}, status=status.HTTP_200_OK)
 
     if request.method == 'PUT':
-        return Response({'msg': 'edit shelter'}, status=status.HTTP_200_OK)
-    
-    if request.method == 'DELETE':
-        return Response({'msg': 'delete shelter'}, status=status.HTTP_200_OK)
+        # Check that the request is from the shelter themselves
+        if request.user == shelter:
+            shelter.email = request.data.get('email', shelter.email)
+            shelter.address = request.data.get('address', shelter.address)
+            shelter.city = request.data.get('city', shelter.city)
+            shelter.province = request.data.get('province', shelter.province)
+            shelter.postal_code = request.data.get('postal_code', shelter.postal_code)
+            shelter.phone = request.data.get('phone', shelter.phone)
+            shelter.avatar = request.data.get('avatar', shelter.avatar)
+            shelter.description = request.data.get('description', shelter.description)
+            shelter.password = request.data.get('password', shelter.password)  # not sure if we should have password?
+
```

Comment:
Add a get is_notif_comment as well in case shelter doesn't want to receive notifications for comments or reviews. Similarly for seeker PR, I don't know what's in there but it should get all three is_notifs_ things